client/dashboard/sites/site-fields/index.tsx

```diff
@@ -35,6 +39,91 @@ function LoadingIndicator( { label }: { label: string } ) {
 	return <TextBlur>{ label }</TextBlur>;
 }
 
+function getSiteManagementUrl( site: Site ) {
+	if ( canManageSite( site ) ) {
+		return `/sites/${ site.slug }`;
+	}
+	return site.options?.admin_url;
+}
+
+const titleFieldStyle = {
+	overflowX: 'hidden',
+	textOverflow: 'ellipsis',
+	whiteSpace: 'nowrap',
+} as const;
+
+export function Name( { site, value }: { site: Site; value: string } ) {
+	const renderBadge = () => {
+		if ( site.is_wpcom_staging_site ) {
+			return <Badge>{ __( 'Staging' ) }</Badge>;
+		}
+
+		if ( isSitePlanTrial( site ) ) {
+			return <Badge>{ __( 'Trial' ) }</Badge>;
+		}
+
+		if ( isP2( site ) ) {
+			return <Badge>{ __( 'P2' ) }</Badge>;
+		}
+
+		return null;
+	};
+
+	return (
+		<Link to={ getSiteManagementUrl( site ) } disabled={ site.is_deleted }>
+			<HStack alignment="center" spacing={ 1 }>
+				{ site.is_deleted ? (
+					<Text variant="muted">{ value }</Text>
```

Comment:
Maybe we should avoid to use Text to keep the font weight for the deleted site as well 🤔

Reply:
But then we need to add a custom styles for the gray color 😭 

I think it's OK if the font weight for deleted sites is different.